### PR TITLE
fix(@cubejs-schema-compiler): remove `ORDER BY` from subqueries

### PR DESCRIPTION
**Pull request by JoshMentzer**
**Check List**
- [x] Tests has been run in packages where changes made if available
- [x] Linter has been run for changed code
- [x] Tests for the changes have been added if not covered yet
- [ ] Docs have been added / updated if required

**Issue Reference this PR resolves**
#657

### CODE REVIEW

File: packages/cubejs-schema-compiler/adapter/MssqlQuery.js [hunk 1]
```diff
@@ -133,6 +133,24 @@ class MssqlQuery extends BaseQuery {
       'date_to'
     )} date_to FROM (VALUES ${values}) ${this.asSyntaxTable} dates (date_from, date_to)`;
   }
+
+  newSubQuery(options) {
+    options = {...options, isSubQuery: true};
```
Comment (paveltiunov) [inline packages/cubejs-schema-compiler/adapter/MssqlQuery.js:138]: Unfortunately, the name is deceiving here. Not only subqueries go through this routine. And also this is really base change and can't go to the MS SQL query.

**Review by paveltiunov — CHANGES_REQUESTED** [review]
@JoshMentzer Hey Joshua! Thanks for contributing to this one! I'd suggest supporting empty order `{}` and adding it here: https://github.com/cube-js/cube.js/blob/835a3c0a9b83c67c5dca6cd0fa4437b249fe6b0d/packages/cubejs-schema-compiler/adapter/BaseQuery.js#L897. This pertains to all databases actually and to not only to MS SQL.

File: packages/cubejs-schema-compiler/adapter/MssqlQuery.js (hunk 1 above)
Comment (JoshMentzer) [reply packages/cubejs-schema-compiler/adapter/MssqlQuery.js:138]: @paveltiunov Totally get it; I kept going back and forth between changing base and adding it to MSSQL when I was working on this; hated the way changing MSSQL 'felt' (repeating the code) and not knowing the code base very well, was reluctant to change Base.
I'll make the change to base and go from there.  What other use case is going through this routine, is it just the PreAggregation stuff?

Comment (paveltiunov) [reply packages/cubejs-schema-compiler/adapter/MssqlQuery.js]: @JoshMentzer Yep. AFAIK it's only pre-aggregations.

**Comment by ovr** [comment]
Oops, very strange issue. I rebased this branch and pushed changes to @JoshMentzer and GitHub closed this PR.

**Comment by ovr** [comment]
Merged in https://github.com/cube-js/cube.js/commit/75c19035e2732adfb7c4711197bba57245e9673e

Thank you @JoshMentzer for your contribution 🍰 🥮

**Pull request closed without merging**